src/app/dashboard/campaign-card/campaign-card.component.ts

```diff
@@ -12,18 +12,19 @@ import { Campaign } from '../dashboard.service';
       <div class="content">
         <h3>
           <a routerLink="{{ '/campaign/' + campaign.id }}">
-            {{ campaign.advertiser && campaign.advertiser.label }}
+            {{ campaign.name }}
           </a>
         </h3>
+        <div>{{ campaign.advertiser && campaign.advertiser.label }}</div>
         <div>
           <span class="status {{ campaign.status }}">{{ campaign.status | titlecase }}</span>
           {{ campaign.type | titlecase }}
         </div>
-        <div *ngIf="campaign.flights | campaignFlightTargets as targets">
+        <div *ngIf="campaign.flights | unique: 'targets':'code' | geoTargets as targets">
```

Comment:
I changed these to chain pipes. First one takes the flight targets or zones and creates a unique set based on the key field. The second pipe is shared with the flight table to convert to comma separated labels for display.

Reply:
+1 nice!  That `unique` pipe is handy!